### PR TITLE
implement media usage tracking for all associations

### PR DESCRIPTION
**Pull request by petrjasek**
not only featuremedia

SDANSA-395

### CODE REVIEW

File: features/links.feature [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        """
+        {"associations": {"pic": {"_id": "two", "type": "picture", "renditions": {}}}}
+        """
+        Then we get OK response
+
+        When we get "links?guid=two"
+        Then we get list with 1 items
+        """
+        {"_items": [
+            {"_id": "one", "type": "text"}
```
Comment (ride90) [inline features/links.feature:27]: The real response will contain a full document(s) not only `_id` and `type`, right?

File: superdesk/metadata/item.py [hunk 2]
```diff
@@ -452,6 +452,31 @@
             }
         }
     },
+
+    # track references to other objects,
+    # based on associations but allows queries
+    'refs': {
```
Comment (ride90) [inline superdesk/metadata/item.py:458]: I think we might need a migration to fill items `refs` based on their existing associations.

File: features/links.feature (hunk 1 above)
Comment (petrjasek) [reply features/links.feature:27]: yep right

File: superdesk/metadata/item.py (hunk 2 above)
Comment (petrjasek) [reply superdesk/metadata/item.py:458]: ansa won't need it as they only care about last 24h, but if anyone else would want that to work for all content ever then yes

**Pull request merged**